Resources/Prototypes/Recipes/Reactions/chemicals.yml

```diff
@@ -1,105 +1,141 @@
 - type: reaction
   id: Ammonia
   reactants:
-    Hydrogen:
-      amount: 3
-    Nitrogen:
-      amount: 1
+  - id: Hydrogen
+    amount: 3
+  - id: Nitrogen
+    amount: 1
   products:
-    Ammonia: 4
+  - id: Ammonia
+    amount: 4
 
 - type: reaction
   id: Diethylamine
   reactants:
-    Ammonia:
-      amount: 1
-    Ethanol:
-      amount: 1
+  - id: Ammonia
+    amount: 1
+  - id: Ethanol
+    amount: 1
   products:
-    Diethylamine: 2
+  - id: Diethylamine
+    amount: 2
 
 - type: reaction
   id: Phenol
   reactants:
-    Water:
-      amount: 1
-    Chlorine:
-      amount: 1
-    Oil:
-      amount: 1
+  - id: Water
+    amount: 1
+  - id: Chlorine
+    amount: 1
+  - id: Oil
+    amount: 1
   products:
-    Phenol: 3
+  - id: Phenol
+    amount: 3
 
 - type: reaction
   id: Acetone
   reactants:
-    Oil:
-      amount: 1
-    WeldingFuel:
-      amount: 1
-    Oxygen:
-      amount: 1
+  - id: Oil
+    amount: 1
+  - id: WeldingFuel
+    amount: 1
+  - id: Oxygen
+    amount: 1
   products:
-    Acetone: 2
+  - id: Acetone
+    amount: 2
+
+- type: reaction
+  id: Cracking
+  rate: 0.1
+  reactants:
+  - id: Oil
+    amount: 1
+  - id: SulfuricAcid
+    amount: 1
+  - id: Oxygen
+    amount: 1
+  - id: Water
+    amount: 1
+    catalyst: true
+  products:
+  - id: Hydrogen
+    amount: 1
+  - id: WeldingFuel
+    amount: 2
+
+# reactants:\W*\n(( {4,6}\w+: *\w* *\n)*)( {4}(\w+): *\n)(( {6}\w+: *\w* *\n)*)
+# reactants:\n$1  - id: $4\n$5
+
+# reactants:\W*\n((( {2}- id: \w+\n)( {6}\w+: *\w* *\n)*)*)( {6}(\w+): *(\w)* *\n)
+# reactants:\n$1    $6: $7\n
+
+# products:\W*\n(( {4}\w+: *\w+ *\n)*)( {4}(\w+): *(\w+) *\n)
+# products:\n$1  - id: $4\n    amount: $5\n
```

Comment:
Regexes for reformatting the reactant/product YAML and a testing reaction. These'll get axed before it's ready to merge.